cmd/prysmctl/withdrawal/withdrawal.go

```diff
@@ -0,0 +1,165 @@
+	"go.opencensus.io/trace"
+)
+
+const (
+	basePath = "/eth/v1"
+	apiPath  = "/beacon/pool/bls_to_execution_changes"
+)
+
+func setWithdrawalAddress(c *cli.Context, r io.Reader) error {
+	ctx, span := trace.StartSpan(c.Context, "withdrawal.blsToExecutionAddress")
+	defer span.End()
+
+	BeaconNodeHost := "http://localhost:3500"
+	if c.String(BeaconHostFlag.Name) != "" {
+		BeaconNodeHost = c.String(BeaconHostFlag.Name)
+	}
+	u, err := url.ParseRequestURI(BeaconNodeHost)
+	if err != nil {
+		return errors.Wrap(err, "invalid format, unable to parse url")
+	}
+	if u.Scheme == "" || u.Host == "" {
+		return fmt.Errorf("url must be in the format of http(s)://host:port url used: %v", BeaconNodeHost)
+	}
+
+	foundFilePaths, err := findWithdrawalFiles(c.String(FileFlag.Name))
+	if err != nil {
+		return err
+	}
+	if len(foundFilePaths) == 0 {
+		return errors.New("no compatible files were found")
+	}
+	au := aurora.NewAurora(true)
+	fmt.Println(au.Red("===============IMPORTANT==============="))
+	fmt.Println(au.Red("Please read the following carefully"))
+	fmt.Println("This action will allow you to partially withdraw any amount over the 32 staked eth in your validator balance. " +
+		"You will also be entitled to the full withdrawal if your validator has exited. " +
+		"Please navigate to the following website and make sure you understand the current implications " +
+		"of changing your bls withdrawal address to an ethereum address. " +
+		"THIS ACTION WILL NOT BE REVERSIBLE ONCE INCLUDED. " +
+		"You will NOT be able to change the address again once changed. ")
+	for _, foundFilePath := range foundFilePaths {
+		b, err := os.ReadFile(filepath.Clean(foundFilePath))
+		if err != nil {
+			return errors.Wrap(err, "failed to open file")
+		}
+		switch string(b)[0:1] {
+		case "[":
+			var to []*apimiddleware.SignedBLSToExecutionChangeJson
+			if err := json.Unmarshal(b, &to); err != nil {
+				return errors.Wrap(err, "failed to unmarshal file")
+			}
+			if len(to) == 0 {
+				return errors.New("the list of signed requests is empty")
+			}
+			for _, jsonOb := range to {
+				if err := verifyWithdrawalCertainty(r, jsonOb); err != nil {
+					return err
+				}
+			}
+			return callWithdrawalEndpoint(ctx, BeaconNodeHost, to)
+		case "{":
+			var to *apimiddleware.SignedBLSToExecutionChangeJson
+			if err := json.Unmarshal(b, &to); err != nil {
+				return errors.Wrap(err, "failed to unmarshal file")
+			}
+			if to == nil || to.Message == nil {
+				return errors.New("the object or object's message field in file is empty")
+			}
+			if err := verifyWithdrawalCertainty(r, to); err != nil {
+				return err
+			}
+			return callWithdrawalEndpoint(ctx, BeaconNodeHost, []*apimiddleware.SignedBLSToExecutionChangeJson{to})
+		default:
+			return errors.New("the provided file is not a json object or list of jason objects")
+		}
+	}
+	return nil
+}
+
+func verifyWithdrawalCertainty(r io.Reader, request *apimiddleware.SignedBLSToExecutionChangeJson) error {
```

Comment:
This will be super annoying for users converting large amounts of addresses, like exchanges. I would not include this.

Reply:
it's a one time only thing, I'll provide something to override, but it worries me if they made a mistake providing this.